src/domain/valueObject/mappingMatchPattern.go

```diff
@@ -4,33 +4,32 @@ import (
 	"errors"
 	"slices"
 	"strings"
+
+	voHelper "github.com/speedianet/os/src/domain/valueObject/helper"
 )
 
 type MappingMatchPattern string
 
 var ValidMappingMatchPatterns = []string{
-	"begins-with",
-	"contains",
-	"equals",
-	"ends-with",
+	"begins-with", "contains", "equals", "ends-with",
 }
 
-func NewMappingMatchPattern(value string) (MappingMatchPattern, error) {
-	value = strings.ToLower(value)
-	if !slices.Contains(ValidMappingMatchPatterns, value) {
-		return "", errors.New("InvalidMappingMatchPattern")
+func NewMappingMatchPattern(value interface{}) (
+	mappingMatchPattern MappingMatchPattern, err error,
+) {
+	stringValue, err := voHelper.InterfaceToString(value)
+	if err != nil {
+		return mappingMatchPattern, errors.New("MappingMatchPatternMustBeString")
 	}
-	return MappingMatchPattern(value), nil
-}
+	stringValue = strings.ToLower(stringValue)
```

Comment:
We could also replace spaces inside the string with - in case someone sends "begins with" for instance.

Reply:
Done!